### PR TITLE
Python exercises

### PR DESCRIPTION
**Pull request by ludwigbothmann**

### CODE REVIEW

**Comment by ludwig-max** [comment]
Python Code Demos for I2ML

=============================
demo-cart

- cell 8: Error because X_test (and y_test) is not defined
- and after that there is missing a lot (like 80% or so) - why?

=============================
demo-kaggle

- there are 2 versions - which one should I look at?

=============================
demo-cmpclass

- cell 10: In the R solution there is no train test split (because that is the default of plot_learner_prediction, see https://mlr3viz.mlr-org.com/reference/plot_learner_prediction.html), but perhaps it is better the way you do it (with train test split), so keep it that way :)
	but explain a bit what you are doing here (most of all the np.meshgrid)
- cell 12: comment a bit more (in general). Here you are plotting train and test data, that is good idea, but say that you do is. When I read that as a student, perhaps I don't have to understand the code in the cell, because it is "just" plotting the data. But without a comment, I don't know that until I read the code line by line, and then I see "ah, ok, that's a plot of the data". And a legend would be nice in the plot.
- cell 13 and following: Could you have all the plots in one cell as in R solution? Now some are missing and since they do not fit on one page I have to scroll up and down to compare them visually. And some are missing, right? (Perhaps, if they do not fill all, you can just use the 5 from the R solution)
- cassini: why are the decision boundaries of the multinomial regression and the LDA not linear? They should be from a theoretical perspective. Is that a bug in the plot_decision_boundaries() function?
- overall: the interpretations are missing
- the last part is missing: why?

=============================
demo-genclass
- You changed a lot of the descriptive text, why?
- nice idea with the 3D plotly plot! :)
- naive Bayes: The "normal" version is missing, correct?
- naive Bayes: The model fit and results of the version wirh categorical predictors is missing, correct?
=> The rest I did not look into that thoroughly yet, but seems pretty nice

=============================
demo-knn
- You changed a lot of the descriptive text, why?
- And there is missing like a lot, isn't it?

=============================
demo-limo
- that one looks really nice. Only in the end, where you have the comparison with scikit-learn and the exact solution: where is the manually computed solution as comparison?

=============================
demo-logreg
- here is missing like the second half, why?

=============================
demo-nested
- here is missing a lot as well?
- plus: undefined "i" in cell 6

=============================
demo-overfitting
- mmce is quite far away from the R solution, why?
- seems pretty complete
- could you add - here and in all other chunks - labels for the confusion matrices? sklearn has truth in rows and predictions in columns, but in I2ML we usually have it the other way around. That's ok as long as the confusion matrices have labels. Thanks :)

=============================
demo-randomforests
- seems pretty complete
- more comments would be nice

=============================
demo-resampling
- seems pretty complete
- why are numbers in cell 14 quite off those from the R solution?

=============================
demo-roc
- seems pretty complete

=============================
demo-splines
- pretty much work in progress?

**Comment by lisa-wm** [comment]
@ludwigbothmann is this still relevant?

**Comment by ludwigbothmann** [comment]
yes

**Comment by ludwigbothmann** [comment]
We have now new, proper python exercises, I close this now and delete the branch.
